contents/dataframesmeta_macros.md

```diff
@@ -0,0 +1,106 @@
+## Macros {#sec:dataframesmeta_macros}
+
+Before we dive into DFM commands,
+you might noticed that all of those start with an "at" `@` symbol.
+These commands have a special category in the Julia language:
+they are called **macros**.
+
+> **_NOTE:_**
+> Macros are a way to perform [metaprogramming](https://en.wikipedia.org/wiki/Metaprogramming) in Julia.
+> We won't cover macro basics here.
+> However, if you want to find more more,
+> please check the [Julia documentation section on metaprogramming](https://docs.julialang.org/en/v1/manual/metaprogramming/)
+> and [Julius Krumbiegel's blog post on Julia macros](https://jkrumbiegel.com/pages/2021-06-07-macros-for-beginners/).
+
+DFM macros behave **similar** as `DataFrames.jl` functions:
+
+- they both take a `DataFrame` as a first positional argument
+- they have in-place _mutating_ functions
+  (as discussed in @sec:df_performance_inplace: the bang `!` functions)
+
+Nevertheless, there are some **differences** on using `DataFrames.jl` functions versus DFM macros.
+
+First, using parentheses in the macro commands are _optional_, and it can be replaced by _spaces_ instead.
+For example:
+
+```julia
+@select(df, :col)
+```
+
+is the same as:
+
+```julia
+@select df :col
+```
+
+We will be using the space syntax in this chapter.
+
+Second, macros parse _static_ commands by default.
+If you want _dynamic_ parsing you'll need to add `$` to your syntax.
+This happens because macros treat the input as a static string.
+For example:
+
+```julia
+@select df :col
+```
+
+will always work because our intended select column command with the argument `:col` won't change in _runtime_
+(the time that the Julia is executing code).
+It will always mean the same operation no matter the context.
+
+Now suppose you want to use one of the column selectors presented in @sec:select.
```

Comment:
```suggestion
Now suppose that you want to use one of the column selectors presented in @sec:select.
```
If I search on Google for "now suppose you", I get redirected mostly to sites mentioning "now suppose that you".